src/tools/wasm2c-wrapper.h

```diff
@@ -79,28 +80,44 @@ u32 (*Z_envZ_getTempRet0Z_iv)(void) = _Z_envZ_getTempRet0Z_iv;
 int main(int argc, char** argv) {
   init();
 
+  // We go through each export and call it, in turn.
+  size_t next = 0;
+
+  // We do this all with a single setjmp. A setjmp is needed to handle wasm
+  // traps, and emitting a single one helps compilation speed into wasm as
+  // compile times are O(size * num_setjmps).
+  while (1) {
+    size_t curr = next;
+    next++;
+    if (setjmp(g_jmp_buf) != 0) {
+      puts("exception!");
+    } else {
+      // Run the next testcase.
+      switch(curr) {
 )";
 
   // For each export in the wasm, emit code to call it and log its result,
   // similar to the other wrappers.
+  size_t exportIndex = 0;
+
   for (auto& exp : wasm.exports) {
     if (exp->kind != ExternalKind::Function) {
       continue;
     }
 
+    ret += "        case " + std::to_string(exportIndex++) + ":\n";
```

Comment:
Would it make sense to use a conventional for loop so `exportIndex` can just be the index variable rather than a separately managed variable?

Reply:
Yeah, I think you're right and that is simpler, fixing.